### ISSUE TITLE
`@IgnoreJRERequirement` annotated class keeps getting "Undefined reference "

### ISSUE THREAD

**Issue by oldergod** [xvik/gradle-animalsniffer-plugin#83]
I have a class with only member methods like

```kotlin
internal class JvmFileSystemFileHandle(
  readWrite: Boolean,
  private val seekableByteChannel: SeekableByteChannel,
) : FileHandle(readWrite) {
```

Animal sniffer would give me violations like:
```
JvmFileSystemFileHandle.class: Undefined reference: java.nio.channels.SeekableByteChannel
JvmFileSystemFileHandle.class:35: Undefined reference: java.nio.channels.SeekableByteChannel java.nio.channels.SeekableByteChannel.truncate(long)
.....
JvmFileSystemFileHandle.class:83: Undefined reference: void java.nio.channels.SeekableByteChannel.close()
```

After having annotated the class as follows:
```kotlin
@org.codehaus.mojo.animal_sniffer.IgnoreJRERequirement
internal class JvmFileSystemFileHandle(
  readWrite: Boolean,
  private val seekableByteChannel: SeekableByteChannel,
) : FileHandle(readWrite) {
```

all violations are gone except one:
```
JvmFileSystemFileHandle.class: Undefined reference: java.nio.channels.SeekableByteChannel
```

Does this need a specific configuration to get ignore of this violation as well?

**Comment by xvik** [comment]
I can only suggest to try inlining import (remove actual import):

```java
@org.codehaus.mojo.animal_sniffer.IgnoreJRERequirement
internal class JvmFileSystemFileHandle(
  readWrite: Boolean,
  private val seekableByteChannel: java.nio.channels.SeekableByteChannel,
) : FileHandle(readWrite) {
```

This way, violation moved in context of ignoring annotation and should disappear.

I saw you publiahed this issue in animalsniffer repository too: maybe, they would be able to tell you more. 
I don't know other ways to workaround.

**Comment by xvik** [comment]
I reproduced your case to make sure it doesn't work. Indeed.
Interesting moment: annotation works properly for **java** class in the same situation. So this must be something specific to kotlin (bytecode generation).

**Comment by xvik** [comment]
Really **bad** workaround is to completely ignore `SeekableByteChannel` in all checked classes: 

```groovy
animalsniffer {
    ignore = ['java.nio.channels.SeekableByteChannel']
}
```

**Comment by oldergod** [comment]
I looked at the decompiled bytecode and tried a few combinations but no luck. Thanks for the help!

**Comment by Marcono1234** [comment]
For future reference, the reason was apparently that the Animal Sniffer dependency of this plugin had not been updated yet, see also discussion in https://github.com/mojohaus/animal-sniffer/issues/249.

**Comment by xvik** [comment]
For completeness: I checked java sample on plugin trunk with 1.23 and it was working, and kotlin sample on examples project, using released plugin version with animalsniffer 1.22.

Workaround: `animalsniffer.toolVersion = '1.23'`

Plugin version 1.7.1 released
